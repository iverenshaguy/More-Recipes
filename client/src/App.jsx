import React from 'react';
import './App.scss';

const App = () => <div className="App">Hi</div>;

export default App;
