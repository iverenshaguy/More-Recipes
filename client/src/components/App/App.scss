@import '../../assets/scss/modules';
@import '~bootstrap/scss/bootstrap';
// sass-lint:disable-block indentation
body {
  &:not(.user-profile) {
    background: $background-gradient, url('../../assets/images/background.jpg') $app-background-color no-repeat fixed;
    background-size: cover;
    margin: 0;
  }
}

a,
.text-green {
  color: $light-green;
}

a {
  &:hover,
  &:focus,
  &:active,
  &.active {
    border: 0;
    box-shadow: none;
    color: $green;
    cursor: pointer;
    outline: none;
    transition: color 0.5s;
  }

  &.dropdown-item {
    &:hover,
    &:focus,
    &:active,
    &.active {
      border: 0;
      background-color: $gray-100;
      box-shadow: none;
      color: $green;
      cursor: pointer;
      outline: none;
      transition: color 0.5s;
    }
  }
}

h4 {
  font-weight: 400;
}

input {
  &:hover,
  &:focus,
  &:active {
    border: 1px solid $light-green;
    outline: none;
    transition: border 0.5s;
  }
}

select {
  &:hover,
  &:focus,
  &:active {
    border: 1px solid $light-green;
    outline: none;
    transition: border 0.5s;
  }
}

form {
  small {
    font-size: 1.05em;
  }
  &-control {
    &:focus {
      border-color: $green;
      box-shadow: $form-box-shadow;
    }
  }
}

.breadcrumb {
  li {
    a {
      color: $navbar-grey;
    }
    &:hover,
    &:focus,
    &:active {
      color: $green;
      transition: color 0.5s;
    }
  }
}

.background-grey {
  background-color: $background-grey;
}

.fa-star,
.fa-star-o {
  color: $fa-gold;
}

.fa-heart,
.fa-heart-o {
  color: $red;
}

.fa-thumbs-up,
.fa-thumbs-o-up {
  color: $fa-gold;
}

.fa-thumbs-down,
.fa-thumbs-o-down {
  color: $navbar-grey;
}

.fa-facebook {
  color: $facebook;
}

.fa-twitter {
  color: $twitter;
}

.fa-pinterest {
  color: $red;
}

.fa-home {
  font-size: 1.5rem;
}

#side-rating-toolbar p i {
  cursor: pointer;
}

.inline-p, .inline-li {
  display: inline !important;
  vertical-align: middle;
}

.btn {
  &:not(.btn-danger) {
    background-color: $light-green;
    border: 1px solid $light-green;
    border-radius: 3px;

    &:hover,
    &:focus,
    &:active,
    &.active {
      background-color: $green;
      border: 1px solid $green;
      color: $white;
      transition: background-color 0.5s;
    }
  }

  &.btn-outline {
    background-color: $white;
    border: 1px solid $green;
    color: $green;
  }

  &.next-previous-btn {
    font-size: 1.2em;
    font-weight: 400;
  }

  &:hover,
  &:focus {
    cursor: pointer;
  }

  li {
    &:hover,
    &:focus,
    &:active,
    &.active {
      background-color: $green;
      border: 1px solid $green;
      color: $white;
      transition: background-color 0.5s;
    }
  }
}

#search {
  border: 1px solid $white;
  border-bottom-left-radius: 5px;
  border-right: 0;
  border-top-left-radius: 5px;
  box-shadow: 0;
  &:focus,
  &:active {
    border-right: 0;
    box-shadow: 0;
  }
}

.search-btn {
  background-color: $white !important;
  border: 1px solid $white !important;
  box-shadow: none !important;
  color: $light-green !important;
  outline: none !important;
  &:hover,
  &:focus,
  &:active {
    border-left: 0;
    color: $green !important;
    cursor: pointer !important;
    transition: color 0.5s;
  }
}

.preview-img-div {
  padding-left: 0;
  padding-right: 0;
}

.photo-preview {
  max-height: 20rem;
  width: 60%;
}

.file-feedback {
  color: $danger-red;
  font-size: 0.875rem;
  margin-top: 0.25rem;
}

// TABS-TOGGLE
.profile-toggle-wrapper {
  border-bottom: 1px solid $recipe-card-grey;

  a {
    font-size: 1.2rem;
    margin-bottom: 0;
    padding-bottom: 0.55rem;
    text-transform: uppercase;
    color: $darkgrey;
    cursor: pointer;

    &.active,
    &:active,
    &:hover,
    &:focus {
      color: $darkgrey;
      text-decoration: none;
      border-bottom: 1.5px solid $green;
    }

    &.back-link {
      margin-right: 1.5rem;
    }

    &.recipes-tab {
      margin-left: 1.5rem;
      margin-right: 1.5rem;
      padding-left: 1rem;
      padding-right: 1rem;
    }

    &.favorites-tab,
    &.categories-tab {
      margin-left: 1.5rem;
      padding-left: 1rem;
      padding-right: 1rem;
    }

    i {
      color: $darkgrey;

      &.fa-heart {
        font-size: 1.4rem;
      }

      &.flaticon:before {
        font-size: 2.3rem;
        margin-left: 0;
      }
    }
  }
}

.close {
  outline: none !important;

  &:hover,
  &:focus,
  &.active {
    outline: none !important;
  }
}

.input-group .form-control {
  margin: 0px !important;
}

// Card Form

.card-form-body {
  &.signup-body {
    margin-top: -30px;
  }

  #card-form-wrapper {
    background-color: $white;
    border: 1px solid $white;
    border-radius: 0.5rem;
    height: auto;
    margin-bottom: 6rem;
  }

  a.next-previous-a {
    // font-size: 1em;
    color: $green;

    &:hover,
    &:focus,
    &.active {
      text-decoration: none;
      font-size: 1.04em;
      transition: font-size 0.2s;
    }
  }

  // form {
  //   button {
  //     margin-top: 2.5rem;
  //   }
  // }
}

.show {
  opacity: 1 !important;
}

@media only screen and (min-width: 1200px) {
  #card-form-wrapper {
    margin-left: 15rem;
    margin-right: 15rem;
  }
}

@media only screen and (min-width: 992px) and (max-width: 1199px) {
  #card-form-wrapper {
    margin-left: 10rem;
    margin-right: 10rem;
  }
}

@media only screen and (min-width: 453px) and (max-width: 575px) {
  #card-form-wrapper {
    margin-left: 2rem;
    margin-right: 2rem;
  }
}

@media only screen and (max-width: 452px) {
  #card-form-wrapper {
    margin-left: 2rem;
    margin-right: 2rem;
  }
}
