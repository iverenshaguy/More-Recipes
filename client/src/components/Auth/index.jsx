import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { Row, Col } from 'reactstrap';
import Signup from './SignupForm';
import Login from './LoginForm';
import './LoginSignup.scss';

/**
 * @exports
 * @param {object} props
 * @returns {component} Auth
 */
const Auth = (props) => {
  const { from } = props.location.state ? props.location.state : { from: { pathname: '/' } };

  if (props.isAuthenticated) {
    return <Redirect to={from} />;
  }

  return (
    <div className="login-signup-body">
      <div className="container" id="login-signup-container">
        <Row className="my-5" id="login-signup-wrapper">
          <Col className="p-xs-2 p-lg-4" id="login-signup-div">
            {props.type === 'signup' ? <Signup {...props} /> : <Login {...props} />}
          </Col>
        </Row>
      </div>
    </div>
  );
};

Auth.propTypes = {
  type: PropTypes.string.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  location: PropTypes.shape({
    hash: PropTypes.string,
    key: PropTypes.string,
    pathname: PropTypes.string,
    search: PropTypes.string,
    state: PropTypes.shape({
      from: PropTypes.shape({
        hash: PropTypes.string,
        key: PropTypes.string,
        pathname: PropTypes.string,
        search: PropTypes.string
      })
    })
  }).isRequired,
};

export { Auth as AuthComponent };

const mapStateToProps = state => ({
  authenticating: state.auth.loading,
  isAuthenticated: state.auth.isAuthenticated,
  submitError: state.auth.error
});

export default connect(mapStateToProps)(Auth);
