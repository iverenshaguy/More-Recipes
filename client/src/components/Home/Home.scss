@import '../../assets/scss/modules';

.home {
  .navbar {
    position: relative;
  }

  #hero-image {
    height: 45vh;
    max-width: 100%;
    position: relative;

    button {
      background-color: $light-green;
      border: 1px solid $light-green;
      color: $white;
      padding: 0.75rem 1.5rem;

      &:hover,
      &:focus,
      &:active {
        background-color: $green;
        border: 1px solid $green;
        color: $white;
      }
    }
  }

  #hero-image-text {
    color: $white;
    padding-top: 5vh;
    text-align: center;
    width: 100%;
  }

  #hero-search-form {
    margin-top: 2rem;
  }
}

#home-add-recipe-btn {
  border-radius: 0;
  box-shadow: 0;
  outline: none;

  &:hover,
  &:focus,
  &:active {
    cursor: pointer;
  }
}
