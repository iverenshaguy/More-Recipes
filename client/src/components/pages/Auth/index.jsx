import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { Row, Col } from 'reactstrap';
import Form from '../../shared/Forms';
import setCurrentLocation from '../../../actions/location';
import { authPropTypes } from '../../../helpers/proptypes';
import './Auth.scss';

/**
 * @exports
 * @class Auth
 * @extends Component
 * @returns {component} Auth
 */
class Auth extends Component {
  static propTypes = {
    ...authPropTypes,
    type: PropTypes.string.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  /**
   * @memberof LoginForm
   * @returns {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('auth'));
  }

  /**
   * @memberof Auth
   * @returns {component} Auth
   */
  render() {
    const { from } = this.props.location.state ? this.props.location.state : { from: { pathname: '/' } };

    if (this.props.isAuthenticated) {
      return <Redirect to={from} />;
    }

    const signupMeta = {
      title: 'Register for a New Account',
      btnText: 'SIGN UP',
      extra: <p className="text-center">Already have an account, <Link to="/login">Log in here</Link>.</p>
    };

    const loginMeta = {
      title: 'Sign In to Your Account',
      btnText: 'SIGN IN',
      extra: <p className="text-center">Dont have an account, <Link to="/signup">Sign up here</Link>.</p>
    };

    return (
      <div className="login-signup-body">
        <div className="container" id="login-signup-container">
          <Row className="my-5" id="login-signup-wrapper">
            <Col className="p-xs-2 p-lg-4" id="login-signup-div">
              {this.props.type === 'signup' ?
                <Form {...this.props} type="signup" meta={signupMeta} /> :
                <Form {...this.props} type="login" meta={loginMeta} />}
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

export { Auth as AuthComponent };

const mapStateToProps = state => ({
  submitting: state.auth.loading,
  isAuthenticated: state.auth.isAuthenticated,
  submitError: state.auth.error
});

export default connect(mapStateToProps)(Auth);
