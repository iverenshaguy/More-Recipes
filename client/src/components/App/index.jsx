import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { ConnectedRouter } from 'react-router-redux';
import 'bootstrap/dist/css/bootstrap.css';
import { history } from '../../reducers';
import Header from '../shared/Header';
import Footer from '../shared/Footer';
import StickyBar from '../shared/StickyBar';
import Modal from '../shared/Modals';
import Home from '../pages/Home';
import Auth from '../pages/Auth';
import Profile from '../pages/Profile';
import FavoriteRecipes from '../pages/FavoriteRecipes';
import Error from '../shared/Error';
import SingleRecipe from '../pages/SingleRecipe';
import { requireAuthentication } from '../hoc';
import '../../assets/font-awesome/css/font-awesome.css';
import '../../assets/kitchen-font/flaticon.css';
import './App.scss';

/**
 * Represents the App Component
 * @returns {component} App
 */
const App = () => (
  <ConnectedRouter history={history}>
    <div className="app">
      <Header />
      <div className="content">
        <Switch>
          <Route exact path="/" component={requireAuthentication(Home)} />
          <Route path="/:username/favorites" component={requireAuthentication(FavoriteRecipes)} />
          <Route path="/:username" component={requireAuthentication(Profile)} />
          <Route exact path="/recipes/:id" component={requireAuthentication(SingleRecipe)} />
          {/* <Route exact path="/recipes" component={requireAuthentication(AllRecipes)} /> */}
          <Route path="/login" render={props => <Auth {...props} type="login" />} />
          <Route path="/signup" render={props => <Auth {...props} type="signup" />} />
          <Route path="/page-not-found" render={() => <Error type={404} />} />
          <Redirect to="/not-found" />
        </Switch>
      </div>
      <Footer />
      <StickyBar />
      <Modal />
    </div>
  </ConnectedRouter>
);

export default App;
