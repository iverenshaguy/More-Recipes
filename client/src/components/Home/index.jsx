import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Hero from './Hero';
import RecipeItems from '../shared/RecipeItems';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  /**
   * @memberof Home
   * @returns {function} updateLocationState
   */
  componentWillMount() {
    this.props.updateLocationState('home');
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero toggle={this.props.toggleModal} />
        <RecipeItems />
      </div>
    );
  }
}

Home.propTypes = {
  toggleModal: PropTypes.func.isRequired,
  updateLocationState: PropTypes.func.isRequired,
};

export default Home;
