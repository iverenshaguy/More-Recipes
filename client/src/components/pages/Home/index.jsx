import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import setCurrentLocation from '../../../actions/location';
import { fetchTopRecipes, searchRecipes } from '../../../actions/recipes';
import Hero from './Hero';
import RecipeItems from '../../shared/RecipeItems';
import PreLoader from '../../shared/PreLoader';
import { multiRecipePropTypes } from '../../../helpers/proptypes';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    isFetching: PropTypes.bool.isRequired,
    ...multiRecipePropTypes
  };

  /**
   * @constructor
   * @memberof Home
   * @returns {nothing} - Returns nothing
   */
  constructor() {
    super();

    this.state = {
      title: 'TOP RECIPES',
      currentPage: 1,
      limit: 5,
      searchValue: ''
    };

    this.handlePageChange = this.handlePageChange.bind(this);
    this.handleSearchInput = this.handleSearchInput.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  /**
   * @memberof Home
   * @return {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('home'));
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    let fetch;
    const { limit, searchValue } = this.state;

    if (this.state.title === 'TOP RECIPES') {
      fetch = fetchTopRecipes(page, limit);
    } else {
      fetch = searchRecipes(searchValue, page, limit);
    }

    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetch));
  }

  /**
   * @memberof Home
   * @param {event} event
   * @returns {state} home
   */
  handleSearchInput(event) {
    this.setState({
      searchValue: event.target.value
    });
  }

  /**
   * @memberof Home
   * @param {object} event
   * @returns {state} home
   */
  handleSearch(event) {
    event.preventDefault();

    const { searchValue, currentPage, limit } = this.state;
    this.setState({
      title: 'SEARCH RESULTS'
    }, () => this.props.dispatch(searchRecipes(searchValue, currentPage, limit)));
  }

  /**
   * @memberof Home
   * @returns {fragment} Recipe items
   */
  renderBody() {
    const { isFetching, recipes, metaData } = this.props;
    const { title } = this.state;

    if (isFetching) {
      return <PreLoader />;
    }

    return (
      <RecipeItems
        title={title}
        recipes={recipes}
        handlePageChange={this.handlePageChange}
        metaData={metaData}
      />
    );
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero
          dispatch={this.props.dispatch}
          handleSearchInput={this.handleSearchInput}
          handleSearch={this.handleSearch}
          searchValue={this.state.searchValue}
        />
        {this.renderBody()}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  recipes: state.recipes.recipes,
  metaData: state.recipes.metaData
});

export { Home as HomeComponent };

export default connect(mapStateToProps)(Home);
