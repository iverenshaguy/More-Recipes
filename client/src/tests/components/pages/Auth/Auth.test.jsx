import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Auth, { AuthComponent } from '../../../../components/pages/Auth';
import Home from '../../../../components/pages/Home';
import App from '../../../../components/App';
import initialValues from '../../../setup/initialValues';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const unAuthStore = mockStore({
  ...initialValues,
  auth: {
    ...initialValues.auth, isAuthenticated: false
  }
});
const authStore = mockStore(initialValues);

const dispatchMock = jest.fn();

const props = {
  submitting: false,
  isAuthenticated: false,
  submitError: null,
  dispatch: dispatchMock
};

const loginLocation = {
  pathname: '/login',
  state: { from: { pathname: '/' } }
};

const signupLocation = {
  pathname: '/signup',
  state: { from: { pathname: '/' } }
};

const signupMeta = {
  title: 'Register for a New Account',
  btnText: 'SIGN UP',
  extra: <p>something</p>
};

const loginMeta = {
  title: 'Sign In to Your Account',
  btnText: 'SIGN IN',
  extra: <p>something</p>
};

const statelessLocation = {
  pathname: '/signup'
};

describe('Auth', () => {
  it('login: renders correctly', () => {
    const shallowWrapper = shallow(<AuthComponent {...props} location={loginLocation} type="login" meta={loginMeta} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('signup: renders correctly', () => {
    const shallowWrapper = shallow(<AuthComponent {...props} location={signupLocation} type="signup" meta={signupMeta} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders login page when unauthenticated', () => {
    const wrapper = mount( //eslint-disable-line
      <MemoryRouter
        initialEntries={['/login']}
      >
        <Provider store={unAuthStore}>
          <App />
        </Provider>
      </MemoryRouter>);

    expect(wrapper.find(Auth)).toHaveLength(1);
    expect(wrapper.find(Home)).toHaveLength(0);

    wrapper.unmount();
  });

  it('redirects to another location when authenticated', () => {
    const wrapper = mount( //eslint-disable-line
      <MemoryRouter
        initialEntries={['/login']}
      >
        <Provider store={authStore}>
          <App />
        </Provider>
      </MemoryRouter>);

    expect(wrapper.find(Home)).toHaveLength(1);
    expect(wrapper.find(Auth)).toHaveLength(0);

    wrapper.unmount();
  });

  it('redirects when authenticated: login', () => {
    const shallowWrapper = shallow(<AuthComponent {...props} isAuthenticated location={loginLocation} type="login" meta={loginMeta} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('redirects when authenticated: signup', () => {
    const shallowWrapper = shallow(<AuthComponent {...props} isAuthenticated location={signupLocation} type="signup" meta={signupMeta} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('redirects when authenticated: stateless location', () => {
    const shallowWrapper = shallow(<AuthComponent {...props} isAuthenticated location={statelessLocation} type="login" meta={loginMeta} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
