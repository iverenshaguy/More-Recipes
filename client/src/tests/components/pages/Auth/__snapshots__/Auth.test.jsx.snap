// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth login: renders correctly 1`] = `
<div
  className="card-form-body"
>
  <div
    className="container"
  >
    <div
      className="my-5"
      id="card-form-wrapper"
    >
      <div
        className="pt-4 pb-3"
        id="login-signup-div"
      >
        <Login
          changeForm={[Function]}
          dispatch={[Function]}
          isAuthenticated={false}
          location={
            Object {
              "pathname": "/login",
              "state": Object {
                "from": Object {
                  "pathname": "/",
                },
              },
            }
          }
          submitError={null}
          submitting={false}
          type="login"
          username={null}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Auth redirects when authenticated: login 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/",
    }
  }
/>
`;

exports[`Auth redirects when authenticated: signup 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "from": Object {
        "pathname": "/username",
      },
    }
  }
/>
`;

exports[`Auth redirects when authenticated: stateless location 1`] = `
<Redirect
  push={false}
  to={
    Object {
      "pathname": "/",
    }
  }
/>
`;

exports[`Auth signup: renders correctly 1`] = `
<div
  className="card-form-body signup-body"
>
  <div
    className="container"
  >
    <div
      className="my-5"
      id="card-form-wrapper"
    >
      <div
        className="pt-4 pb-3"
        id="login-signup-div"
      >
        <Signup
          changeForm={[Function]}
          dispatch={[Function]}
          isAuthenticated={false}
          location={
            Object {
              "pathname": "/signup",
              "state": Object {
                "from": Object {
                  "pathname": "/",
                },
              },
            }
          }
          submitError={null}
          submitting={false}
          type="signup"
          username={null}
        />
      </div>
    </div>
  </div>
</div>
`;
